### PR TITLE
Note that invite links are invalidated after upgrading to 3.9

### PR DESCRIPTION
**Pull request by jasonblais**
Already noted in the 3.9 changelog.

It also takes upstream changes from the `master` branch, hence the larger diff file.

### CODE REVIEW

File: source/administration/upgrade.md [hunk 1]
```diff
@@ -65,6 +63,8 @@ To start, select one of the following guides:
       1. Run `platform -version` to check the current version of your Mattermost server
       2. Determine the appropriate next upgrade for your server:
           - Mattermost `v3.0.x` and later can upgrade directly to the [latest release of Mattermost](https://about.mattermost.com/download/).
+              - Note: Upgrading to `v3.9.x` and later will invalidate existing team invite links, password reset links and email verifications.
```
Comment (JeffSchering) [inline source/administration/upgrade.md]: See previous comment.

File: source/administration/upgrade.md [hunk 2]
```diff
@@ -11,19 +11,17 @@ To start, select one of the following guides:
...
 
 **To upgrade your system**:
 
 1. Download the **appropriate next upgrade** of your Team Edition server and note any compatibility procedures:
       1. Run `platform -version` to check the current version of your Mattermost server
       2. Determine the appropriate next upgrade for your server:
-          - Mattermost `v3.0.x` and later can upgrade directly to the [latest release of Mattermost](https://about.mattermost.com/download/), but see the previous note about upgrading from 3.7.x and earlier.
+          - Mattermost `v3.0.x` and later can upgrade directly to the [latest release of Mattermost](https://about.mattermost.com/download/) considering the conditions below:
+              - Note: Upgrading to `v3.9.x` and later will invalidate existing team invite links, password reset links and email verifications.
```
Comment (JeffSchering) [inline source/administration/upgrade.md]: Need a comma after "password reset links"

What happens when email verifications are invalidated? Do sys admins need to take any action, or be aware of any consequences? If so, you should mention them here.

Comment (jasonblais) [inline source/administration/upgrade.md]: @JeffSchering It's actually `email verification links`, not verifications. Clarified

File: source/administration/upgrade.md [hunk 3]
```diff
@@ -11,19 +11,17 @@ To start, select one of the following guides:
...
       1. Run `platform -version` to check the current version of your Mattermost server
       2. Determine the appropriate next upgrade for your server:
-          - Mattermost `v3.0.x` and later can upgrade directly to the [latest release of Mattermost](https://about.mattermost.com/download/), but see the previous note about upgrading from 3.7.x and earlier.
+          - Mattermost `v3.0.x` and later can upgrade directly to the [latest release of Mattermost](https://about.mattermost.com/download/) considering the conditions below:
+              - Note: Upgrading to `v3.9.x` and later will invalidate existing team invite links, password reset links, and email verification links.
+              - Note: Upgrading to `v3.8.x` and later, if you're using NGINX as a proxy for the Mattermost Server, replace the `location /api/v3/users/websocket {` line with `location ~ /api/v[0-9]+/(users/)?websocket$ {` in the `/etc/nginx/sites-available/mattermost` NGINX configuration file. [See documentation to learn more](https://docs.mattermost.com/install/install-ubuntu-1404.html#configuring-nginx-as-a-proxy-for-mattermost-server).
```
Comment (esethna) [inline source/administration/upgrade.md]: Is this supposed to be 3.8 or 3.9 and later?

File: source/administration/upgrade.md [hunk 4]
```diff
@@ -65,6 +63,8 @@ To start, select one of the following guides:
       1. Run `platform -version` to check the current version of your Mattermost server
       2. Determine the appropriate next upgrade for your server:
           - Mattermost `v3.0.x` and later can upgrade directly to the [latest release of Mattermost](https://about.mattermost.com/download/).
+              - Note: Upgrading to `v3.9.x` and later will invalidate existing team invite links, password reset links, and email verification links.
+              - Note: Upgrading to `v3.8.x` and later, if you're using NGINX as a proxy for the Mattermost Server, replace the `location /api/v3/users/websocket {` line with `location ~ /api/v[0-9]+/(users/)?websocket$ {` in the `/etc/nginx/sites-available/mattermost` NGINX configuration file. [See documentation to learn more](https://docs.mattermost.com/install/install-ubuntu-1404.html#configuring-nginx-as-a-proxy-for-mattermost-server).
```
Comment (esethna) [inline source/administration/upgrade.md]: Same comment here

File: source/administration/upgrade.md (hunk 3 above)
Comment (jasonblais) [inline source/administration/upgrade.md]: It's 3.8 - I submitted a separate PR so that it's already included in the existing upgrade guide: https://github.com/mattermost/docs/pull/1144
